src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/Frame.cs

```diff
@@ -1370,9 +1372,9 @@ private IPipe CreateRequestBodyPipe()
             => ConnectionInformation.PipeFactory.Create(new PipeOptions
             {
                 ReaderScheduler = ServiceContext.ThreadPool,
-                WriterScheduler = ConnectionInformation.InputWriterScheduler,
-                MaximumSizeHigh = ServiceContext.ServerOptions.Limits.MaxRequestBufferSize ?? 0,
-                MaximumSizeLow = ServiceContext.ServerOptions.Limits.MaxRequestBufferSize ?? 0
+                WriterScheduler = InlineScheduler.Default,
+                MaximumSizeHigh = 1,
+                MaximumSizeLow = 1
```

Comment:
@davidfowl Are you saying that ideally the request body Pipe would buffer instead of the connection "Input" Pipe?